pkgs/applications/misc/xmrig/default.nix

```diff
@@ -1,20 +1,36 @@
-{ stdenv, lib, fetchFromGitHub, cmake, libuv, libmicrohttpd, openssl
+{ stdenv, lib, fetchFromGitHub, cmake, hwloc, libuv, libmicrohttpd, openssl
 , donateLevel ? 0
+, http ? false
+, tls ? true
+, algo ? ["cryptonight-lite" "cryptonight-heavy" "cryptonight-pico" "cryptonight-gpu" "randomx"]
 }:
 
 stdenv.mkDerivation rec {
   name = "xmrig-${version}";
-  version = "2.14.4";
+  version = "3.0.0";
 
   src = fetchFromGitHub {
     owner = "xmrig";
     repo = "xmrig";
     rev = "v${version}";
-    sha256 = "0hmla9ys9gmmcrmkidw05ygiz734rs7skb9w86jnwxi5nlpxqrhj";
+    sha256 = "1m0rsjb7y1j77mzg5cqb3fdvzgvjkrwgmkjn9nv1xl2757z8hcl4";
   };
 
+  cmakeFlags = []
+    ++ lib.optional (!http) "-DWITH_HTTP=OFF"
+    ++ lib.optional (!tls) "-DWITH_TLS=OFF"
+    ++ lib.optional ((lib.count(a: a == "cryptonight-lite")  algo) == 0) "-DWITH_CN_LITE=OFF"
+    ++ lib.optional ((lib.count(a: a == "cryptonight-heavy") algo) == 0) "-DWITH_CN_HEAVY=OFF"
+    ++ lib.optional ((lib.count(a: a == "cryptonight-pico")  algo) == 0) "-DWITH_CN_PICO=OFF"
+    ++ lib.optional ((lib.count(a: a == "cryptonight-gpu")   algo) == 0) "-DWITH_CN_GPU=OFF"
+    ++ lib.optional ((lib.count(a: a == "randomx")           algo) == 0) "-DWITH_RANDOMX=OFF"
```

Comment:
It would be better if this would fail for unknown `algo` values, so that typos would result in errors instead of silently being ignored.

Also, I think explicitly passing `=ON` or `=OFF` would be better in case upstream defaults change in the future.